### PR TITLE
Fix display of 0 for UpDownWidget (bootstrap-4 and material-ui)

### LINKED ISSUE #2042: Material-UI theme problem with value 0 (zero) in UpDownWidget
Dear authors, the "initial 0 number" problem was fixed recently in #1795 except that the fix there didn't catch the UpDownWidget. That one still tests for truthy here: https://github.com/rjsf-team/react-jsonschema-form/blob/master/packages/material-ui/src/UpDownWidget/UpDownWidget.tsx#L43 and that line needs a similar change into `value={value || (value === 0) ? value : ''}` 
Could you fix that one, too, please? Thanks a lot!

### Steps to Reproduce

[playground link](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJudW1iZXIiOjAsImludGVnZXIiOjAsIm51bWJlckVudW0iOjIsImludGVnZXJSYW5nZSI6NDIsImludGVnZXJSYW5nZVN0ZXBzIjo4MH0sInNjaGVtYSI6eyJ0eXBlIjoib2JqZWN0IiwidGl0bGUiOiJOdW1iZXIgZmllbGRzICYgd2lkZ2V0cyIsInByb3BlcnRpZXMiOnsibnVtYmVyIjp7InRpdGxlIjoiTnVtYmVyIiwidHlwZSI6Im51bWJlciJ9LCJpbnRlZ2VyIjp7InRpdGxlIjoiSW50ZWdlciIsInR5cGUiOiJpbnRlZ2VyIn0sIm51bWJlckVudW0iOnsidHlwZSI6Im51bWJlciIsInRpdGxlIjoiTnVtYmVyIGVudW0iLCJlbnVtIjpbMSwyLDNdfSwibnVtYmVyRW51bVJhZGlvIjp7InR5cGUiOiJudW1iZXIiLCJ0aXRsZSI6Ik51bWJlciBlbnVtIiwiZW51bSI6WzEsMiwzXX0sImludGVnZXJSYW5nZSI6eyJ0aXRsZSI6IkludGVnZXIgcmFuZ2UiLCJ0eXBlIjoiaW50ZWdlciIsIm1pbmltdW0iOjQyLCJtYXhpbXVtIjoxMDB9LCJpbnRlZ2VyUmFuZ2VTdGVwcyI6eyJ0aXRsZSI6IkludGVnZXIgcmFuZ2UgKGJ5IDEwKSIsInR5cGUiOiJpbnRlZ2VyIiwibWluaW11bSI6NTAsIm1heGltdW0iOjEwMCwibXVsdGlwbGVPZiI6MTB9fX0sInVpU2NoZW1hIjp7ImludGVnZXIiOnsidWk6d2lkZ2V0IjoidXBkb3duIn0sIm51bWJlckVudW1SYWRpbyI6eyJ1aTp3aWRnZXQiOiJyYWRpbyIsInVpOm9wdGlvbnMiOnsiaW5saW5lIjp0cnVlfX0sImludGVnZXJSYW5nZSI6eyJ1aTp3aWRnZXQiOiJyYW5nZSJ9LCJpbnRlZ2VyUmFuZ2VTdGVwcyI6eyJ1aTp3aWRnZXQiOiJyYW5nZSJ9fSwidGhlbWUiOiJtYXRlcmlhbC11aSIsImxpdmVTZXR0aW5ncyI6eyJ2YWxpZGF0ZSI6ZmFsc2UsImRpc2FibGUiOmZhbHNlLCJvbWl0RXh0cmFEYXRhIjpmYWxzZSwibGl2ZU9taXQiOmZhbHNlfX0=)  where the "Number" field correctly has the zero, but the "Integer" field is blank where we would have expected the zero, too.

#### Expected behavior

Both "Number" and "Integer" field should show the value zero `0`

#### Actual behavior

The "Integer" field is blank

### Version

Current playground

### PR DESCRIPTION
**Pull request by wegry**
https://github.com/rjsf-team/react-jsonschema-form/issues/2042

### Reasons for making this change

We render data from an endpoint that contains numeric 0s. When we load up the data in @rjsf/bootstrap-4, values that come back from the server as `0` are `''` in inputs. Updating an existing model currently breaks `0` field values by setting them to `null`.

fixes #2042

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### Deploy preview

Once CI finishes, you should be able to view a deploy preview of the playground from this PR at this link: https://deploy-preview-[PR_NUMBER]--rjsf.netlify.app/

### CODE REVIEW

**Review by jimmycallin — APPROVED** [review]

File: packages/bootstrap-4/src/UpDownWidget/UpDownWidget.tsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ const UpDownWidget = ({
         type="number"
         disabled={disabled}
         readOnly={readonly}
-        value={value ? value : ""}
+        value={value != null ? value : ""}
```
Comment (epicfaace) [inline packages/bootstrap-4/src/UpDownWidget/UpDownWidget.tsx:41]: Let's do this for clarity and consistency with the other themes (I believe fluent-ui):

```suggestion
        value={value || value === 0 ? value : ""}
```

**Review by epicfaace — COMMENTED** [review]
Thanks for this change! Can you also do a quick pass across all the other themes to see if this problem exists in the other themes' UpDownWidget component, and if so, fix them too?

Thanks!

File: packages/bootstrap-4/src/UpDownWidget/UpDownWidget.tsx (hunk 1 above)
Comment (wegry) [reply packages/bootstrap-4/src/UpDownWidget/UpDownWidget.tsx]: How would you feel about nullish coalescing? `value ?? ''`? I think my tweaks to the babel build should transpile it to something like this.

Comment (wegry) [reply packages/bootstrap-4/src/UpDownWidget/UpDownWidget.tsx]: I take that back, not so in the unit tests...

File: packages/material-ui/src/UpDownWidget/UpDownWidget.tsx [hunk 2]
```diff
@@ -40,7 +40,7 @@ const UpDownWidget = ({
         required={required}
         type="number"
         disabled={disabled || readonly}
-        value={value ? value : ''}
+        value={value != null ? value : ''}
```
Comment (epicfaace) [inline packages/material-ui/src/UpDownWidget/UpDownWidget.tsx:43]: this one too

**Comment by wegry** [comment]
@epicfaace 

> Can you also do a quick pass across all the other themes to see if this problem exists in the other themes' UpDownWidget component, and if so, fix them too?

Is there a different widget name similar to UpDownWidget? Otherwise, I believe I only saw them in fluent, bootstrap-4, and material-ui. They should all be using the same `value || value === 0 ? value : ''` now.

**Comment by epicfaace** [comment]
> @epicfaace
> 
> > Can you also do a quick pass across all the other themes to see if this problem exists in the other themes' UpDownWidget component, and if so, fix them too?
> 
> Is there a different widget name similar to UpDownWidget? Otherwise, I believe I only saw them in fluent, bootstrap-4, and material-ui. They should all be using the same `value || value === 0 ? value : ''` now.

Great, we should be good now! Thanks.

**Review by epicfaace — APPROVED** [review]

**Pull request merged**